homeassistant/components/knx/climate.py

```diff
@@ -213,8 +213,8 @@ def __init__(self, knx_module: KNXModule, config: ConfigType) -> None:
             f"{self._device.temperature.group_address_state}_"
             f"{self._device.target_temperature.group_address_state}_"
             f"{self._device.target_temperature.group_address}_"
-            f"{self._device._setpoint_shift.group_address}"  # noqa: SLF001
-        )
```

Comment:
Looks like a bug in ruff, because the noqa got moved

Reply:
https://github.com/astral-sh/ruff/issues/15375

Reply:
You are faster than me this time :stuck_out_tongue: Hmm, with that we will need to wait until 0.9.1, I think :thinking:

Reply:
I mean, it was a breaking change and it looked like something would change and that sounds interesting :P

Reply:
Yeah, you probably want to wait for 0.9.1. The only alternative is disabling formatting for this statement, which seems undesirable. I hope to get a release out later today, depending on if https://github.com/astral-sh/ruff/pull/15378 gets accepted.

Reply:
I mean there is not that big of a rush now, all the other formatting changes are already being handled in separate PRs, so in the end this PR will be small

Reply:
I just released Ruff 0.9.1 which contains a fix. It won't undo the change itself. You have to move the comment back to its original position but Ruff should then keep it there.